src/phpDocumentor/Descriptor/Traits/CanHaveAType.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @link https://phpdoc.org
+ */
+
+namespace phpDocumentor\Descriptor\Traits;
+
+use phpDocumentor\Descriptor\Interfaces\InheritsFromElement;
+use phpDocumentor\Descriptor\IsTyped;
+use phpDocumentor\Reflection\Type;
+
+use function array_filter;
+use function trigger_error;
+
+use const E_USER_DEPRECATED;
+
+trait CanHaveAType
+{
+    /** @var Type|null $type normalized type of this argument */
+    protected ?Type $type = null;
+
+    public function setType(?Type $type): void
+    {
+        $this->type = $type;
+    }
+
+    public function getType(): ?Type
```

Comment:
Should type be optional? 
- in case of an argument, when it is missing it can be mixed?
- in case of an function / method return type the result is void? (This comes from the PSR-19 spec AFAIK)

Reply:
This is a really good question. There are more elements in the descriptors for which we can ask: should this be nullable. Given that I do not exactly what will break when I remove the nullability from this element I have chosen to postpone addressing this issue :)

Concerning this whole effort: I have mentally prepared myself that I/we need to go over each Descriptor and trait several times to improve it. This is intended to prevent too many monster pr and isolate potentially dangerous changes in a single PR :)